### PR TITLE
completed

### PR DESCRIPTION
**Pull request by bug-smith**
completed.

### CODE REVIEW

**Review by RobertGardner — APPROVED** [review]
Nice

File: api-authorization/server/lib/authorization-middleware.ts [hunk 1]
```diff
@@ -0,0 +1,46 @@
+/* eslint-disable @typescript-eslint/no-unused-vars -- Remove me */
+import { NextFunction, Request, Response } from 'express';
+import jwt, { JwtPayload } from 'jsonwebtoken';
+import { ClientError } from './client-error.js';
+
+interface PayLoad extends JwtPayload {
```
Comment (RobertGardner) [inline api-authorization/server/lib/authorization-middleware.ts]: This type should already exist. It's declared in auth.d.ts.

File: api-authorization/server/lib/authorization-middleware.ts [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+  next: NextFunction
+) {
+  const auth = req.get('authorization');
+  const token = auth?.split('Bearer ')[1];
+  const secretToken = process.env.TOKEN_SECRET;
+  if (!secretToken) throw new ClientError(401, 'Auth failed');
+  if (!auth || !token) {
+    throw new ClientError(401, 'Auth required');
+  }
+  jwt.verify(token, secretToken, (err, decoded) => {
```
Comment (RobertGardner) [inline api-authorization/server/lib/authorization-middleware.ts]: Don't use the callback version. Just get the returned value from `jwt.verify()` and use it.

File: api-authorization/server/lib/authorization-middleware.ts [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+    if (err) {
+      throw new ClientError(401, 'Auth required');
+    }
+    const payload = decoded as PayLoad;
+    req.user = payload;
```
Comment (RobertGardner) [inline api-authorization/server/lib/authorization-middleware.ts]: The best way to take advantage of the types in auth.d.ts is:
`req.user = decoded as Request['user'];`

File: api-authorization/server/server.ts [hunk 4]
```diff
@@ -0,0 +1,195 @@
...
+    const token = jwt.sign(payload, hashKey);
+    res.json({ token, user: payload });
+  } catch (err) {
+    next(err);
+  }
+});
+
+app.get('/api/entries', authMiddleware, async (req, res, next) => {
+  try {
+    if (!req.user) {
```
Comment (RobertGardner) [inline api-authorization/server/server.ts]: You can delete this. `req.user` will always exist when this function executes as long as you are using `authMiddleware`.

**Review by RobertGardner — CHANGES_REQUESTED** [review]
Sorry, I looked at the wrong code before. There is an issue here I'd like you to fix. (The one about not using the callback in jwt.verify.)

File: api-authorization/server/lib/authorization-middleware.ts (hunk 2 above)
Comment (bug-smith) [reply api-authorization/server/lib/authorization-middleware.ts]: Wow, this revision really made the code look so much prettier, thank you!

File: api-authorization/server/server.ts [hunk 5]
```diff
@@ -88,9 +88,6 @@ app.post('/api/auth/sign-in', async (req, res, next) => {
 
 app.get('/api/entries', authMiddleware, async (req, res, next) => {
   try {
-    if (!req.user) {
```
Comment (RobertGardner) [inline api-authorization/server/server.ts]: There are several other places where you could make the same change.

**Review by RobertGardner — APPROVED** [review]

**Pull request merged**